### PR TITLE
Improve abi types in wasmv1

### PR DESCRIPTION
**Pull request by bilboquet**
Related to
[issue 246 from massa-sc-runtime](https://github.com/massalabs/massa-sc-runtime/issues/246)

### CODE REVIEW

**Comment by bilboquet** [comment]
We have to merge soon, so I'm opening the PR to sync with the API guys and prevent conflict as much as possible on both sides.

**Comment by bilboquet** [comment]
On the abi side we estimate that we won't have many more changes for the moment.
I think it worth merging into main so we can synchronize with the api side on shared types etc. 
So feel free to comment :)

File: proto/commons/massa/model/v1/commons.proto [hunk 1]
```diff
@@ -34,3 +34,19 @@ message SecureShare {
...
+// Comparison result
+message ComparisonResult {
+  // Left is lower
+  message LeftLower {}
+  // Left is equal to right
+  message Equal {}
+  // Left is greater
+  message LeftGreater {}
+  // Status
+  oneof restult {
```
Comment (damip) [inline proto/commons/massa/model/v1/commons.proto:47]: out of curiosity why not an Enum to avoid having to declare 3 types ?

Comment (damip) [reply proto/commons/massa/model/v1/commons.proto:47]: General rule: when there are multiple options, enum unless some of those options need to be complex types, then use OneOf

Comment (damip) [reply proto/commons/massa/model/v1/commons.proto:47]: typo: restult

Comment (bilboquet) [reply proto/commons/massa/model/v1/commons.proto]: done :)

Comment (damip) [reply proto/commons/massa/model/v1/commons.proto]: cool ! can you check if it was also done elsewhere ?

Comment (bilboquet) [reply proto/commons/massa/model/v1/commons.proto]: I'm currently applying modifications to other repositories.
I'll also check scheme previously used for ComparisonResult does not appear elsewhere

Comment (bilboquet) [reply proto/commons/massa/model/v1/commons.proto]: Haven't found any other "creative" use of OneOf in the .protos present in this branch ^^

File: proto/abis/massa/abi/v1/abi.proto [hunk 2]
```diff
@@ -175,8 +175,11 @@ message RespResult {
     SubNativeAmountResult sub_native_amount_result = 55;
     TransferCoinsResult transfer_coins_result = 56;
     UnsafeRandomResult unsafe_random_result = 57;
-    VerifyEvmSigResult verify_evm_sig_result = 58;
+    EvmVerifySigResult evm_verify_sig_result = 58;
```
Comment (bilboquet) [inline proto/abis/massa/abi/v1/abi.proto:178]: I can't believe it!
It's not sorted!!!

File: proto/abis/massa/abi/v1/abi.proto [hunk 3]
```diff
@@ -175,8 +175,11 @@ message RespResult {
...
     VerifySigResult verify_sig_result = 59;
+    EvmGetAddressFromPubkeyResult evm_get_address_from_pubkey_result = 60;
+    EvmGetPubkeyFromSignatureResult evm_get_pubkey_from_signature_result = 61;
+    IsAddressEoaResult is_address_eoa_result = 62;
```
Comment (bilboquet) [inline proto/abis/massa/abi/v1/abi.proto:182]: what's the meaning of this one?

**Review by Eitu33 — APPROVED** [review]

**Review by Leo-Besancon — APPROVED** [review]
Approved despite the lack of sorting..

**Pull request merged**